File: app/src/main/java/com/example/acplite/adapters/RecyclerProductsAdapter.java
package com.example.acplite.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.acplite.R;
import com.example.acplite.entidades.Arbol;
import com.example.acplite.entidades.Productos;
import com.example.acplite.helpers.ItemTapListener;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;

public class RecyclerProductsAdapter extends RecyclerView.Adapter<RecyclerProductsAdapter.ViewHolder>{

    private ArrayList<Productos> productosList;
    private final ItemTapListener mTapListener;

    public RecyclerProductsAdapter(ArrayList<Productos> productosList, ItemTapListener mTapListener) {
        this.productosList = productosList;
        this.mTapListener = mTapListener;
    }

    @NonNull
    @NotNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull @NotNull ViewGroup parent, int viewType) {
        RecyclerProductsAdapter.ViewHolder viewHolder = new RecyclerProductsAdapter.ViewHolder(
                LayoutInflater.from(parent.getContext()).inflate(R.layout.item_products, parent, false),
                mTapListener);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull @NotNull ViewHolder holder, int position) {
        holder.txtID.setText("ID: " + String.valueOf(productosList.get(position).getProductID()));
        holder.txtNombre.setText("Nombre: " + productosList.get(position).getProductName());
        holder.txtPrecio.setText("Precio: " + productosList.get(position).getProductPrice());
        holder.imgProducts.setImageBitmap(productosList.get(position).getProductImg());
    }

    @Override
    public int getItemCount() {
        return productosList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        ImageView imgProducts;
        TextView txtID, txtNombre, txtPrecio;

        public ViewHolder(@NonNull @NotNull View itemView, ItemTapListener mTapListener) {
            super(itemView);
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    mTapListener.onItemTap(v, getAdapterPosition());
                }
            });
            imgProducts = itemView.findViewById(R.id.ImgProduct);
            txtID = itemView.findViewById(R.id.txtProductID);
            txtNombre = itemView.findViewById(R.id.txtProductName);
            txtPrecio = itemView.findViewById(R.id.txtProductPrice);
        }
    }
}

File: app/src/main/java/com/example/acplite/adapters/RecyclerProductsAdapter.kt
package com.example.acplite.adapters

import com.example.acplite.entidades.Productos
import com.example.acplite.helpers.ItemTapListener
import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import com.example.acplite.R
import android.widget.TextView
import java.util.ArrayList

class RecyclerProductsAdapter(
    private val productosList: ArrayList<Productos>,
    private val mTapListener: ItemTapListener
) : RecyclerView.Adapter<RecyclerProductsAdapter.ViewHolder>() {
    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        return ViewHolder(
            LayoutInflater.from(parent.context).inflate(R.layout.item_products, parent, false),
            mTapListener
        )
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.txtID.text = "ID: " + productosList[position].productID.toString()
        holder.txtNombre.text = "Nombre: " + productosList[position].productName
        holder.txtPrecio.text = "Precio: " + productosList[position].productPrice
        holder.imgProducts.setImageBitmap(productosList[position].productImg)
    }

    override fun getItemCount(): Int {
        return productosList.size
    }

    inner class ViewHolder(itemView: View, mTapListener: ItemTapListener) :
        RecyclerView.ViewHolder(itemView) {
        var imgProducts: ImageView
        var txtID: TextView
        var txtNombre: TextView
        var txtPrecio: TextView

        init {
            itemView.setOnClickListener { v -> mTapListener.onItemTap(v, adapterPosition) }
            imgProducts = itemView.findViewById(R.id.ImgProduct)
            txtID = itemView.findViewById(R.id.txtProductID)
            txtNombre = itemView.findViewById(R.id.txtProductName)
            txtPrecio = itemView.findViewById(R.id.txtProductPrice)
        }
    }
}